[2025.06.10] 프로젝트 디렉토리 초기 셋업

seonghunlee94 opened the pull request.

seonghunlee94 committed:
[2025.06.10] 프로젝트 디렉토리 초기 셋업

heli-os commented on project/seonghunlee-onboarding/build.gradle:
https://course4.slack.com/archives/C08UM9WS117/p1749288050063589

참고하셔서 작업해주세요.

seonghunlee94 committed:
[2025.06.17] 프로젝트 디렉토리 초기 셋업

seonghunlee94 committed:
[2025.06.17] 프로젝트 디렉토리 초기 셋업

seonghunlee94 commented:
이렇게 초기 세팅하면 되는 걸까요?

seonghunlee94 committed:
설문조사 ERD 작성

seonghunlee94 committed:
[answer 유효성 체크] 단답형 유효성 검사 구현

1. 단답형 기준
- 띄어쓰기 2번 이상일 경우X
- 공백이 연속 2칸인 경우X
- 글자수 10글자까지 단답형
- 특수문자 포함X
- 양쪽 공백 무시

seonghunlee94 committed:
[answer 유효성 체크] 장문형 유효성 검사 구현

1. 장문형 기준
- 글자수 최소 20글자, 최대 100글자
- 연속 공백 3칸 이상X
- 특수문자 일부 허용(느낌표, 물결, 점, 콤마)

seonghunlee94 committed:
[answer 유효성 체크] 단일 선택 리스트 유효성 검사 구현

1. 단일 선택 리스트 기준
- null은_유효한_값이다 > 필수 여부인 경우, 다른 객체 통해 체크할 예정
- 선택지에_있는_값이면_유효함
- 선택지에_없는_값이면_유효하지_않음

seonghunlee94 committed:
[answer 유효성 체크] 다중 선택 리스트 구현 시, 매개변수 타입을 다양화하기 위해 제네릭으로 변경

seonghunlee94 committed:
[answer 유효성 체크] 단일 선택 리스트에서 선택 목록 없는 경우 유효하도록 구현

seonghunlee94 committed:
[answer 유효성 체크] 다중 선택 리스트 유효성 검사 구현

1. 다중 선택 리스트 기준
- 허용된 선택지만 포함되었을 때 유효함
- 유효하지 않은 값이 포함되면 유효하지_않음
- 선택지가 비어있으면 모든 입력은 유효함
- 입력값이 null이면 유효함
- 입력값이_빈리스트이면_유효함

seonghunlee94 committed:
디렉토리 구조 변경

seonghunlee94 committed:
[answer 유효성 체크] 필수값 체크를 위한 RequiredValidator 추가 및 입력 형식 체크하는 클래스로 위임하도록 로직 구현

seonghunlee94 committed:
Merge pull request #285 from INNER-CIRCLE-ICD4/seonghunlee-onboarding-answer-validator

응답 형식 체크 및 필수값 체크 구현했습니다.

interface를 통해 객체 지향적으로 클래스를 만들어 체크하도록 구현해보고 싶었습니다.

그리고 필수값 체크의 경우, 필수값 체크되면 응답 형식 체크하는 클래스들로 위임하도록 구현해보려고 했는데 이런 방식이 적절한 고민인 지 궁금합니다.
